### PR TITLE
Add associations

### PR DESCRIPTION
**Pull request by Ngufor-emmanuella**
### In this milestone of the project I added the following classes ;

1. Created a class Classroom
2. Created the has-many/belongs-to relationship between Classroom and Student
3. Created a class Book
4. Created a class Rental
5. Create the many-to-many relationship between Person and Book using the intermediate class Rental

### CODE REVIEW

File: books.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class Books
+  attr_accessor :title, :author
+  attr_reader :rentals
+
```
Comment (omar-muhamad) [inline books.rb]: - [ ] Please add Getters and Setters (attr_accessor) for `@rentals`, not Getters only (attr_reader) as you did.

File: books.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+    rental.book = self
+  end
```
Comment (omar-muhamad) [inline books.rb]: - [ ] This method suppose to create rentals and the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association, So please fix that.
[Hint] the method will be like that:
```ruby
def add_rental(person, date) 
    Rental.new(date, self, person)
end
```

File: person.rb [hunk 3]
```diff
@@ -2,13 +2,14 @@
 
 class Person < Nameable
   attr_accessor :name, :age
-  attr_reader :id
+  attr_reader :id, :rentals
```
Comment (omar-muhamad) [inline person.rb]: - [ ] Same here, Please add Getters and Setters (attr_accessor) for `@rentals`, not Getters only (attr_reader) as you did.

File: person.rb [hunk 4]
```diff
@@ -25,4 +26,9 @@ def correct_name
   def of_age?
     @age >= 18
   end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+    rental.person = self
+  end
```
Comment (omar-muhamad) [inline person.rb]: - [ ] Same here,  This method suppose to create rentals and the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association, So please fix that.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @Ngufor-emmanuella,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by m05e5 — APPROVED** [review]
**Hi @Ngufor-emmanuella 

You did great. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
CleverScornfulBuffalo-size_restricted

## Highlights
✔️ Descriptive PR 
✔️ your linters are all ok

## Optional suggestions


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**m05e5**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._**

**Pull request merged**